### DISCUSSION TITLE
Return type for data_dump.py and parsing issue (SMS reading)

### DISCUSSION THREAD

**Discussion by enterth** [Salamek/huawei-lte-api#174] [Q&A] [answered]
Hello,

First thanks for your work.

I have a question about data_dump.py (example).

I'm able to dump data and get this format :

`==== Sms.get_sms_list
{'Count': '5',
 'Messages': {'Message': [{'Content': 'bla bla bla',
                           'Date': '2023-02-08 15:17:00',
                           'Index': '40005',
                           'Phone': '+XXXXXXXXXXX',
                           'Priority': '0',
                           'SaveType': '4',
                           'Sca': None,
                           'SmsType': '1',
                           'Smstat': '0'},
                          {'Content': 'Follow the white rabbit',
                           'Date': '2023-02-08 15:08:40',
                           'Index': '40004',
                           'Phone': '+XXXXXXXXXXX',
                           'Priority': '0',
                           'SaveType': '4',
                           'Sca': None,
                           'SmsType': '1',
                           'Smstat': '0'}]}}`

I processed data with PHP & removed "==== Sms.get_sms_list" , changed single quote to double quote and changed "'Sca': None," to ""Sca': false"" to get a valid JSON file.

It was working until i got multiple line SMS.

`{'Count': '5',
 'Messages': {'Message': [{'Content': 'bla bla bla'
                                      'bla bla bla'
                                      'bla bla bla',
                           'Date': '2023-02-08 15:17:00',
                           'Index': '40005',
                           'Phone': '+XXXXXXXXXXX',
                           'Priority': '0',
                           'SaveType': '4',
                           'Sca': None,
                           'SmsType': '1',
                           'Smstat': '0'},
                          {'Content': 'Follow the white rabbit',
                           'Date': '2023-02-08 15:08:40',
                           'Index': '40004',
                           'Phone': '+XXXXXXXXXXX',
                           'Priority': '0',
                           'SaveType': '4',
                           'Sca': None,
                           'SmsType': '1',
                           'Smstat': '0'}]}}`

Does anyone have a solution to process this file correctly? I would like to extract with python & send data to a webhook that i coded in PHP, i tried to identify the type of format of this return but didn't find it.

(I can work on the detection & correction of the file but i think it's a waste of time)

Any help would me really apprecied 🙏

Thanks

**Comment by Salamek** [comment, marked as answer]
Output of data_dump.py and co. is not JSON it is `pprint` format of Python `dict` (similar type to PHP `array`) ... If you want your output to be JSON you need to change this line:

https://github.com/Salamek/huawei-lte-api/blob/8c8b269e36c8f63d7fdb7343e1848c5ddc3cdf64/examples/data_dump.py#L27

To:
```python
        print(json.dumps(method()))
```
and add `import json` to imports on start of data_dump.py


And BTW there is a PHP port of `huawei-lte-api` https://github.com/icetee/huawei-lte-api-php (but looks like it is not updated much so IDK if it will work for you)

Anyway if you want to stick with Python lib, i would suggest you to write small python service that will pull required data and push it to your REST endpoint (cron+requests POST should do it) instead of calling data_dump.py and trying to process it in PHP...

**Reply by enterth** [reply]
@Salamek Thanks, it works perfectly. (I'm not much into python but will work on this one, thanks for the php link i'll take a look too).
